petri/src/vm/hyperv/vm.rs

```diff
@@ -75,35 +78,46 @@ impl HyperVVM {
     }
 
     /// Get the name of the VM
-    pub fn get_name(&self) -> &str {
+    pub fn name(&self) -> &str {
         &self.name
     }
 
     /// Get the VmId Guid of the VM
-    pub fn get_vmid(&self) -> &Guid {
+    pub fn vmid(&self) -> &Guid {
         &self.vmid
     }
 
     /// Get Hyper-V logs for the VM
-    pub fn get_logs(&self) -> anyhow::Result<Vec<String>> {
+    fn event_logs(&self) -> anyhow::Result<Vec<powershell::WinEvent>> {
         let mut logs = Vec::new();
         for log_name in [
             "Microsoft-Windows-Hyper-V-Worker-Admin",
             "Microsoft-Windows-Hyper-V-VMMS-Admin",
         ] {
             logs.append(&mut powershell::run_get_winevent(
                 log_name,
-                self.create_time,
+                &self.create_time,
                 &self.vmid.to_string(),
             )?);
         }
         Ok(logs)
     }
 
-    /// Write logs
-    pub fn write_logs(&self) -> anyhow::Result<()> {
-        for line in self.get_logs()? {
-            self.log_file.write_entry(line);
+    /// Get Hyper-V logs and write them to the log file
+    pub fn flush_logs(&self) -> anyhow::Result<()> {
+        for event in self.event_logs()? {
+            self.log_file.write_entry_fmt(
+                match event.level {
+                    1 | 2 => Level::ERROR,
+                    3 => Level::WARN,
+                    5 => Level::TRACE,
+                    _ => Level::INFO,
+                },
+                format_args!(
+                    "[{}] {}: ({}, {}) {}",
+                    event.time_created, event.provider_name, event.level, event.id, event.message,
```

Comment:
If you can parse `time_created` into a jiff timestamp, it would be really useful to propagate that through to `JsonLog::write_entry`. Then our upcoming log visualizer can sort by timestamp and get these events back in the right order, rather than have all these events show up at the end of the test or whatever.

Reply:
(Maybe have `LogWriter` have an `Option<Timestamp>` in it or whatever)

Reply:
Looking at the output now... ideally this would also have more than 1-second granularity, otherwise the sorting is not going to be as useful.